lib/ascii_binder/helpers.rb

```diff
@@ -118,8 +123,47 @@ def build_config
       validate_config(YAML.load_stream(open(build_config_file)))
     end
 
+    def create_new_repo
+      gem_template_dir = File.expand_path("../../../templates", __FILE__)
+
+      # Create the new repo dir
+      begin
+        Dir.mkdir(source_dir)
+      rescue Exception => e
+        raise "Could not create directory '#{source_dir}': #{e.message}"
+      end
+
+      # Copy the basic repo content into the new repo dir
+      Find.find(gem_template_dir).each do |path|
+        next if path == gem_template_dir
+        src_path = Pathname.new(path)
+        tgt_path = src_path.sub(gem_template_dir,source_dir)
+        if src_path.directory?
+          FileUtils.mkdir_p(tgt_path.to_s)
+        else
+          FileUtils.cp src_path.to_s, tgt_path.to_s
+        end
+      end
+
+      # Initialize the git repo and check everything in
+      g = nil
+      begin
+        g = Git.init(source_dir)
+      rescue Exception => e
+        raise "Could not initialize git repo in '#{source_dir}': #{e.message}"
+      end
+      g.add(:all => true)
+      g.commit("Initial check-in")
```

Comment:
The problem with _not_ having a commit is that the other operations will fail until at least one commit exists. But on the other hand, if the user doesn't have any git config set up, that could lead to trouble, too. Okay - sounds like for the non-`create` actions I'll need to test for the presence of a commit before proceeding.